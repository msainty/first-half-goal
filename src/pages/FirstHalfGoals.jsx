// @flow
import React, { Component } from 'react';
import { connect, type Dispatch } from 'react-redux';
import styled from 'styled-components';
import Fixture from '@components/Fixture';
import type { State as ReduxState } from '../redux/modules';
import type { Action, Fixture as FixtureType } from '../redux/modules/firsthalfgoals/types';
import { fetchFixtureRequest } from '../redux/modules/firsthalfgoals/actions';
import { getFixtures, getNoFixtures } from '../redux/modules/firsthalfgoals/selectors';

const PageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

type Props = {
  fixtures: Array<FixtureType>,
  fetchFixtures: () => void,
  hasNoFixtures: boolean,
};

class FirstHalfGoals extends Component<Props> {
  componentDidMount() {
    const { fetchFixtures } = this.props;
    fetchFixtures();
  }

  render() {
    const { fixtures, hasNoFixtures } = this.props;
    return (
      <PageContainer>
        {hasNoFixtures && 'No fixtures today'}
        {!hasNoFixtures
        && Array.isArray(fixtures) && fixtures.map((fixture) => (
          <Fixture key={fixture.uuid} fixture={fixture} />
        ))}
      </PageContainer>
    );
  }
}

const mapStateToProps = (state: ReduxState) => ({
  hasNoFixtures: getNoFixtures(state.firstHalfGoals),
  fixtures: getFixtures(state.firstHalfGoals),
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  fetchFixtures: () => dispatch(fetchFixtureRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(FirstHalfGoals);
