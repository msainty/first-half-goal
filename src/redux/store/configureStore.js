// @flow
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { rootReducer, rootSaga } from '../modules';
import { sagaMiddleware } from '../enhancers/middleware';

export default function configureStore(initialState: any) {
  const middlewares = [thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares, sagaMiddleware);
  const enhancers = [middlewareEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);
  const store = createStore(rootReducer, initialState, composedEnhancers);
  sagaMiddleware.run(rootSaga);
  return store;
}
