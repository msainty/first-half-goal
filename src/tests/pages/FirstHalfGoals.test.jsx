import React from 'react';
import { Provider } from 'react-redux';
// import moment from 'moment';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import FirstHalfGoals from '@pages/FirstHalfGoals';
import initialState from '../redux/initialState';

const middlewares = [];
const mockStore = configureStore(middlewares);

it('renders with no fixtures', () => {
  const store = mockStore(initialState);

  const tree = renderer
    .create(<Provider store={store}><FirstHalfGoals /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with fixtures', () => {
  const mockState = {
    ...initialState,
    firstHalfGoals: {
      fixtures: [{
        uuid: '1',
        kickOff: '2019-01-02',
        country: 'England',
        league: 'PL',
        status: 0.6,
        homeTeam: 'Manchester United',
        awayTeam: 'Brighton',
        played: 15,
        noOfNoScoreFirstHalf: 0,
        percentageOfAFirstGoalHome: 0.66,
        percentageOfAFirstGoalAway: 0.55,
        probability: 0.85,
        price: 1.35,
        matched: 344,
        rating: 5,
      }],
    },
  };
  const store = mockStore(mockState);
  const tree = renderer
    .create(<Provider store={store}><FirstHalfGoals /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
