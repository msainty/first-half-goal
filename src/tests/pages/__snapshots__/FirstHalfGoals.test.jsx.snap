// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with fixtures 1`] = `
<div
  className="sc-htpNat dePyzG"
>
  <div
    className="sc-bdVaJa bsHmJl"
  >
    <div
      className="sc-bwzfXH czactv"
    >
      2019-01-02
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      England
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      PL
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      Manchester United
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      Brighton
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      15
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      0
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      0.66
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      0.55
    </div>
    <div
      className="sc-bwzfXH czactv"
    >
      0.85
    </div>
  </div>
</div>
`;

exports[`renders with no fixtures 1`] = `
<div
  className="sc-htpNat dePyzG"
>
  No fixtures today
</div>
`;
